### PR TITLE
Enabled skipped tests caused by running the suites on a fresh cloud

### PR DESCRIPTION
**Pull request by patrick-tolosa**
- Add timeouts to ensure files have a chance to update
- Fix an implementation bug on the wait() function
- Move wait() to a utility file
- Create a sample.jpg file when creating a new cloud
- Refactor complex failing tests to improve stability

### CODE REVIEW

File: test/api_spec.js [hunk 1]
```diff
@@ -927,7 +929,7 @@ describe("api", function () {
         public_id: 'test_folder1/test_subfolder2/item',
         tags: UPLOAD_TAGS,
       }),
-    ]).then(function (results) {
+    ]).then(wait(3000)).then(function (results) {
```
Comment (asisayag2) [inline test/api_spec.js:932]: Why do we need to wait here? 
If the upload returned, the assets supposed to be there

File: test/api_spec.js [hunk 2]
```diff
@@ -1008,17 +1013,16 @@ describe("api", function () {
       return uploadImage({
         public_id: PUBLIC_ID_BACKUP_1,
         backup: true,
-      }).then(() => uploadImage({
+      }).then(wait(1000)).then(() => uploadImage({
```
Comment (asisayag2) [inline test/api_spec.js:1016]: Not sure I get why running on a fresh cloud impacts the time we need to wait until we do something. 
Can you explain?

File: test/archivespec.js [hunk 3]
```diff
@@ -176,7 +176,7 @@ describe("archive", function () {
           sinon.assert.calledWith(write, sinon.match(helper.uploadParamMatcher("target_format", "zip")));
         });
       });
-      it.skip('should create archive with "zip" format and include multiple resource types', function () {
+      it('should create archive with "zip" format and include multiple resource types', function () {
```
Comment (asisayag2) [inline test/archivespec.js:179]: This test still fails in all envs

File: test/uploader_spec.js [hunk 4]
```diff
@@ -560,7 +560,7 @@ describe("uploader", function () {
       expect(result.moderation[0].kind).to.eql("manual");
     });
   });
-  it.skip("should support requesting ocr analysis", function () {
+  it("should support requesting ocr analysis", function () {
```
Comment (asisayag2) [inline test/uploader_spec.js:563]: Don't see any change here, how come it suddenly working?

File: test/api_spec.js (hunk 1 above)
Comment (patrick-tolosa) [reply test/api_spec.js:932]: We've encountered again and again that it's not the case.

There seems to be some very small, but non-zero delay from the upload success to the assets returning from an API call. 

We didn't encounter these issues as the account already had these assets in place

File: test/archivespec.js (hunk 3 above)
Comment (patrick-tolosa) [reply test/archivespec.js:179]: Worked locally, I've noticed a few more failed on travis, will review.

File: test/uploader_spec.js (hunk 4 above)
Comment (patrick-tolosa) [reply test/uploader_spec.js:563]: This is due to an account configuration, the account needed an OCR package.

Comment (asisayag2) [reply test/uploader_spec.js:563]: Got it

File: test/api_spec.js (hunk 2 above)
Comment (patrick-tolosa) [reply test/api_spec.js:1016]: No definitive answer I'm afraid.

It's either these tests were never 100% stable and this timeout increases their stability,
or maybe the fresh cloud still runs some background things that gives a slight delay to assets being uploaded.

Comment (asisayag2) [reply test/api_spec.js:1016]: Don't like voodoos.
I suggest checking with someone from the BE to see if your assumption makes sense

Comment (patrick-tolosa) [reply test/api_spec.js]: This was confirmed by the backend team, while they don't know for a fact what can cause the increased latency for a new cloud, we definitely feel it.

File: test/api_spec.js (hunk 1 above)
Comment (asisayag2) [reply test/api_spec.js:933]: Didn't we say will create a retry mechanism? or you want to keep for another task? if you do, please create a ticket.

Comment (patrick-tolosa) [reply test/api_spec.js:933]: Yes it'll be another ticket to resolve these, we have a lot of them all over the code.

**Pull request merged**